src/plugins/Trace.ts

```diff
@@ -63,20 +65,45 @@ export class Trace {
         this.tracer.destroy();
         const { originalContext, originalEvent, reporter } = data;
 
-        this.rootSpan = this.tracer._startSpan(originalContext.functionName, {
-            rootTraceId: this.pluginContext.traceId,
-            domainName: DomainNames.API,
-            className: ClassNames.LAMBDA,
-        });
-
         // awsRequestId can be `id` or undefined in local lambda environments, so we generate a unique id here.
         if (!originalContext.awsRequestId || originalContext.awsRequestId === 'id') {
             originalContext.awsRequestId = Utils.generateId();
         }
 
-        this.pluginContext.transactionId = originalContext.awsRequestId;
-        this.pluginContext.traceId = Utils.generateId();
-        this.pluginContext.spanId = this.rootSpan.spanContext.spanId;
+        const propagatedSpanContext: ThundraSpanContext =
+            this.extractSpanContext(originalEvent) as ThundraSpanContext;
+
+        if (propagatedSpanContext) {
+            this.pluginContext.transactionId = Utils.getConfiguration(
+                envVariableKeys.THUNDRA_LAMBDA_TRACE_USE_PROPAGATED_TRANSACTION_ID) === 'true' ?
+                originalContext.awsRequestId : propagatedSpanContext.transactionId;
+
+            this.pluginContext.traceId = propagatedSpanContext.traceId;
+            this.pluginContext.spanId = propagatedSpanContext.spanId;
+
+            this.tracer.transactionId = this.pluginContext.transactionId;
+
+            this.rootSpan = this.tracer._startSpan(originalContext.functionName, {
+                rootTraceId: this.pluginContext.traceId,
+                domainName: DomainNames.API,
+                className: ClassNames.LAMBDA,
+            });
+
+            this.rootSpan.spanContext.parentId = propagatedSpanContext.spanId;
```

Comment:
Can't this be done by passing span context to `_startSpan` here? (Passing `traceId`, `transactionId`, `parentId` and `baggageItems` at once) So we don't need to modify span context after created it

Reply:
Done, added `parentContext` option to `_startSpan ` method.